stglib/tb.py

```diff
@@ -0,0 +1,249 @@
+        elif "Measure Parameters" in row:
+            header["MeasureParameters"] = col[2]
+        elif "Test Code" in row:
+            header["TestCode"] = col[2]
+        elif "Scan Type" in row:
+            header["ScanType"] = col[2]
+        elif "Periods" in row:
+            header["Periods"] = col[1]
+        elif "Memory Wrap" in row:
+            header["MemoryWrap"] = col[2]
+        elif "Broadcast Normal" in row:
+            header["BroadcastNormalLog"] = col[3]
+        elif "Broadcast Alarm" in row:
+            header["BroadcastAlarmLog"] = col[3]
+        elif "Alarm Scan" in row:
+            header["AlarmScan"] = col[2]
+        elif "Scanned Upto" in row:
+            header["ScannedUpto"] = col[3]
+        elif "Firmware Version" in row:
+            header["FirmwareVersion"] = col[2]
+        elif "TruWare" in row:
+            header["TruWareVersion"] = col[1]
+
+    f.close()
+    return header
+
+
+def read_csv(filnam, skiprows=20, encoding="utf-8"):
+    """Read data from a TruBlue 255 .csv file into an xarray
+    Dataset.
+    """
+
+    df = pd.read_csv(
+        filnam,
+        skiprows=skiprows,
+        header=None,
+        names=["ID", "Name", "Address", "time", "Elapsed", "P_1", "T_28"],
+        encoding=encoding,
+        index_col=False,
+    )
+    df["time"] = df["time"].str.strip("'")
+    df["time"] = df["time"].astype("datetime64[ns]")
+    df.set_index("time", inplace=True)
+    tb = df.to_xarray()
+    return tb
+
+
+def txt_to_cdf(metadata):
+    """
+    Load a raw .txt file and generate a .cdf file
+    """
+    basefile = metadata["basefile"]
+
+    # Read header info from file
+    header = read_header(basefile + ".csv")
+
+    # Add attribute sample_mode continuous to use RBR nc2waves.py
+    header["sample_mode"] = "CONTINUOUS"
+
+    # Append to metadata variable
+    metadata.update(header)
+
+    # Read raw data
+    ds = read_csv(basefile + ".csv", skiprows=metadata["skiprows"])
+
+    # Convert pressure from psi to dbar
+    ds["P_1"] = ds.P_1 / 14.503773800722 * 10
+
+    metadata.pop("skiprows")
+
+    ds = utils.write_metadata(ds, metadata)
+
+    ds = utils.ensure_cf(ds)
+
+    cdf_filename = ds.attrs["filename"] + "-raw.cdf"
+
+    ds.to_netcdf(cdf_filename, unlimited_dims=["time"])
+
+    print(f"Finished writing data to {cdf_filename}")
+
+    return ds
+
+
+def cdf_to_nc(cdf_filename, atmpres=None):
+    """
+    Load a raw .cdf file and generate a processed .nc file
+    """
+
+    # Load raw .cdf data
+    ds = xr.open_dataset(cdf_filename)
+
+    # remove units in case we change and we can use larger time steps
+    ds.time.encoding.pop("units")
+
+    # Add sample_interval to metadata
+    tdelta = timedelta(
+        hours=int(ds.attrs["Periods"][3:5]),
+        minutes=int(ds.attrs["Periods"][6:8]),
+        seconds=float(ds.attrs["Periods"][9:]),
+    )
+    ds.attrs["sample_interval"] = tdelta.total_seconds()
+
+    # Drop unneeded variables
```

Comment:
Put these in a list instead of several calls. Also, best practice is to use drop_vars instead of drop

Reply:
Changed